requirements/build.txt

```diff
@@ -0,0 +1,633 @@
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    pip-compile --generate-hashes --output-file=all.txt base.in dev.in gui-dev.in gui.in
```

Comment:
> Great work! 💯 Sorry I wasn't able to help out. I took a glance at the actions docs after you filed the bug, but confess I found myself a little lost and intimidated. Haven't had much experience with build tools apart from sr.ht.

No worries! This is quite the beast. Thanks for taking a look.

> A lot of the black formatting changes seem to be purely because of some function calls/dicts/tuples having trailing commas, often in disagreement to their siblings. Not really worth going to the effort of fixing now though.

Yeah, I think this is basically just noise. If I had the time, I would probably break out the black stuff separately for clarity, but I've already put more into this than I probably should.

> Hah, can't comment directly on empty files: I take it from the commit message the pytest.ini is needed for a test to run? The docs [mention](https://docs.pytest.org/en/stable/customize.html#pytest-ini)
> 
> > pytest.ini files take precedence over other files, even when empty.
> 
> but don't seem to elaborate on what effect an empty file actually has (apart from determining the rootdir ?)

I think this might just be work in progress that I forgot to delete. Will check before landing.

> > Also, Mac builds don't work on MacOS 11, I will probably skip publishing a new version for Mac until the underlying issues are fixed (see [pyinstaller/pyinstaller#5107](https://github.com/pyinstaller/pyinstaller/issues/5107), which has links to a number of the problems).
> 
> Would it not be worth still doing a macOS build and just including a "not for 11" disclaimer? Or is take-up that good in macOS land that there aren't that many non-11 users left?

I honestly don't know. I know QA uses mozregression all the time to test on older versions of MacOS, so (thinking about it) it's probably worth keeping builds going even if they don't work on 11 yet. I can probably find someone inside Mozilla to test for me (I personally am on 11 already).

> I take it Windows builds aren't also switched to Actions because it's not needed as appveyor is still working/not limited unlike Travis?

Yeah, I figured we could handle Windows later.